src/Styleguide/Utils/Responsive.tsx

```diff
@@ -34,20 +34,80 @@ export class ResponsiveProvider extends React.Component<
 > {
   constructor(props) {
     super(props)
+    const breakpointKeys = Object.keys(props.breakpoints)
+
+    // Build initial breakpoint map --> { breakpoint1: false, breakpoint2: false, ...etc}
+    let breakpoints = breakpointKeys
+      .map(breakpoint => ({
+        [breakpoint]: breakpoint === props.initialBreakpoint ? true : false,
+      }))
+      .reduce((acc, curr) => ({ ...acc, ...curr }), {})
+
+    // Build up the MediaQueryList objects that observe mq changes
+    const mediaMatchers = this.setupMatchers(props.breakpoints, breakpointKeys)
+
+    // Perform initial breakpoint check so that first render is correct
+    breakpoints = this.checkBreakpoints(
+      breakpoints,
+      breakpointKeys,
+      mediaMatchers
+    )
+
     this.state = {
-      breakpoints: {
-        ...Object.keys(props.breakpoints)
-          .map(breakpoint => ({
-            [breakpoint]: breakpoint === props.initialBreakpoint ? true : false,
-          }))
-          .reduce((acc, curr) => ({ ...acc, ...curr }), {}),
-      },
-      mediaMatchers: [],
-      mqHandler: null,
+      breakpoints,
+      breakpointKeys,
+      mediaMatchers,
+    }
+  }
+
+  /**
+   * Create an array of media matchers that can validate each breakpoint
+   */
+  setupMatchers = (breakpoints, breakpointKeys) => {
+    return breakpointKeys.map(breakpoint =>
+      window.matchMedia(breakpoints[breakpoint])
+    )
+  }
+
+  /**
+   * Uses the mediaMatchers list to build a map of the states of each breakpoint
+   */
+  checkBreakpoints = (breakpoints, breakpointKeys, mediaMatchers) => {
```

Comment:
In case you’re taking these in as parameters for ease of testing purposes, then I’d suggest making it a standalone function and type the params. Otherwise as a method I’d suggest just accessing `this.state.*` directly because it means you’ll automatically get the right types.